### PR TITLE
FIX: allow single-string color for scatter

### LINKED ISSUE #12429: scatter() does not accept gray strings anymore
### Bug report

**Bug summary**

Passing a particula gray intensity as a string seems to be broken in matplotlib 3.0, at least for the scatter() function.
This kind of argument format should (and was) possible according to documentation.

**Code for reproduction**

```python
plt.scatter(np.arange(10), np.arange(10), c="0.5")
```

**Actual outcome**

```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-99-426550bf20a7> in <module>()
----> 1 plt.scatter(np.arange(10), np.arange(10), c="0.5")

~/anaconda3/lib/python3.6/site-packages/matplotlib/pyplot.py in scatter(x, y, s, c, marker, cmap, norm, vmin, vmax, alpha, linewidths, verts, edgecolors, data, **kwargs)
   2791         x=x, y=y, s=s, c=c, marker=marker, cmap=cmap, norm=norm,
   2792         vmin=vmin, vmax=vmax, alpha=alpha, linewidths=linewidths,
-> 2793         verts=verts, edgecolors=edgecolors, data=data, **kwargs)
   2794     sci(__ret)
   2795     return __ret

~/anaconda3/lib/python3.6/site-packages/matplotlib/__init__.py in inner(ax, data, *args, **kwargs)
   1783                         "the Matplotlib list!)" % (label_namer, func.__name__),
   1784                         RuntimeWarning, stacklevel=2)
-> 1785             return func(ax, *args, **kwargs)
   1786 
   1787         inner.__doc__ = _add_data_doc(inner.__doc__,

~/anaconda3/lib/python3.6/site-packages/matplotlib/axes/_axes.py in scatter(self, x, y, s, c, marker, cmap, norm, vmin, vmax, alpha, linewidths, verts, edgecolors, **kwargs)
   4187             try:  # First, does 'c' look suitable for value-mapping?
   4188                 c_array = np.asanyarray(c, dtype=float)
-> 4189                 n_elem = c_array.shape[0]
   4190                 if c_array.shape in xy_shape:
   4191                     c = np.ma.ravel(c_array)

IndexError: tuple index out of range
```

**Expected outcome**
 same as:
```python
plt.scatter(np.arange(10), np.arange(10), c=(0.5,0.5,0.5))
```

**Matplotlib version**
  * Operating system: Linux
  * Matplotlib version: 3.0
  * Matplotlib backend (`print(matplotlib.get_backend())`): notebook/inline
  * Python version: 3.6 (conda, default)

### LINKED ISSUE #12438: Scatter doesn't accept a list of strings as color spec.
_Originally posted by @ImportanceOfBeingErnest in https://github.com/matplotlib/matplotlib/pull/12431#issuecomment-427697109_

I wonder if this is the correct fix. The documentation also states

> `c` : color, sequence, or sequence of color, optional, default: 'b'
The marker color. Possible values:
>
>* A single color format string.
>* **A sequence of color specifications of length n.**

So for 

    plt.scatter([0,1],[2,3],c=("0.5", "0.7"))

one would expect two points with different shades of grey. But this is converted to an array of values and shown with the viridis colormap.  
Not sure if the documentation is wrong here or the code.

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

Closes #12429
Closes #12438

New test fails master, passes this PR, and looks correct:

```python 
import matplotlib.pyplot as plt
import numpy as np

fig, ax = plt.subplots()
ax.scatter(np.arange(10), np.arange(10), c='0.5')
plt.show()
```

scatter

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by jklymak** [comment]
Travis error is MacOS image tolerance of 0.005 changing for `test_constrained_layout.py::test_colorbar_location`.  Not sure what changed on macOS to cause this, but it doesn't affect this PR.

**Comment by ImportanceOfBeingErnest** [comment]
I wonder if this is the correct fix. The documentation also states 

> `c` : color, sequence, or sequence of color, optional, default: 'b'
The marker color. Possible values:
>
>* A single color format string.
>* **A sequence of color specifications of length n.**

So for 

    plt.scatter([0,1],[2,3],c=("0.5", "0.7"))

one would expect two points with different shades of grey. But this is converted to an array of values and shown with the viridis colormap.  
Not sure if the documentation is wrong here or the code.

**Comment by jklymak** [comment]
The newest two commits allow lists like `['0.5', '0.7', '0.9', '0.5']` which I agree the docs say we should be able to do.

File: setup.py [hunk 1]
```diff
@@ -15,13 +15,12 @@
     error = """
 Matplotlib 3.0+ does not support Python 2.x, 3.0, 3.1, 3.2, 3.3, or 3.4.
 Beginning with Matplotlib 3.0, Python 3.5 and above is required.
-
```
Comment (QuLogic) [inline setup.py]: You seem to have some extra changes in this file.

File: lib/matplotlib/tests/test_axes.py [hunk 2]
```diff
@@ -1763,6 +1765,9 @@ def test_scatter_color(self):
         ([0.5]*3, None),  # should emit a warning for user's eyes though
         ([0.5]*4, None),  # NB: no warning as matching size allows mapping
         ([0.5]*5, "shape"),
+        # list of strings:
+        (['0.5', '0.4', '0.6', '0.7'], None),
+        (['0.5', 'red', '0.6', 'C5'], None),
```
Comment (timhoffm) [inline lib/matplotlib/tests/test_axes.py]: Can you please add mixtures of string and float to the test? I.e. `c=[0.5, '0.5']` and `c=['0.5', 0.5]`.

Not sure what this should result in, but it should either be a defined output or raise an appropriate error. Either way it should be tested.

Comment (jklymak) [reply lib/matplotlib/tests/test_axes.py]: Good idea.  Done!

Comment (jklymak) [reply lib/matplotlib/tests/test_axes.py]: Note that it errors, which I think is correct.

**Comment by jklymak** [comment]
ping @QuLogic to clear block (setup.py was inadvertently modified)

**Pull request merged**